src/modules/notifSettings/components/NotificationSettingsBody.tsx

```diff
@@ -0,0 +1,105 @@
+import { FC, useEffect, useState } from 'react';
+
+import { Box } from 'blocks';
+
+import { ModalResponse } from 'common';
+
+import { useAccount } from 'hooks';
+
+import { ChannelDashboardNullState } from 'modules/channelDashboard/components/ChannelDashboardNullState';
+import { ChannelSetting } from 'modules/channelDashboard/ChannelDashboard.types';
+
+import { NotificationSettingsLists } from './NotificationSettingsLists';
+import { NotificationSettingsFooter } from './NotificationSettingsFooter';
+import { NotificationSettingsModal } from './NotificationSettingsModal';
+
+type NotificationSettingsBodyProps = {
+  modalControl: ModalResponse;
+  channelSettings: ChannelSetting[];
+  loadingSettings: boolean;
+  settingsToEdit: ChannelSetting;
+  setSettingsToEdit: (settingsToEdit: ChannelSetting) => void;
+};
+
+const NotificationSettingsBody: FC<NotificationSettingsBodyProps> = ({
+  modalControl,
+  channelSettings,
+  loadingSettings,
+  settingsToEdit,
+  setSettingsToEdit,
+}) => {
+
+  const { isWalletConnected, connect } = useAccount();
+  const [newSettings, setNewSettings] = useState<ChannelSetting[]>([]);
+
+  const { open } = modalControl;
+
+  useEffect(() => {
+    if (channelSettings && !loadingSettings) {
+      setNewSettings(channelSettings);
+    }
+  }, [loadingSettings]);
+
+  const handleDeleteSetting = (settingToDelete: ChannelSetting) => {
+    setNewSettings((settings) => settings.filter((setting) => setting.index !== settingToDelete.index));
+  };
+
+  const handleSettingsChange = (newSetting: ChannelSetting) => {
+    const index = newSettings.findIndex((setting) => setting.index === newSetting.index);
+    if (index === -1) setNewSettings([...newSettings, newSetting]);
+    else {
+      const updatedSetting = [...newSettings];
+      updatedSetting[index] = newSetting;
```

Comment:
```suggestion
      const updatedSetting = newSettings.map((setting, settingIndex)=>settingIndex===index? {...newSetting} : setting);
```

Test this out before pushing.